concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -378,14 +378,14 @@ data HashedPersistentBlockState pv = HashedPersistentBlockState {
 
 -- |Convert a 'PersistentBlockState' to a 'HashedPersistentBlockState' by computing
 -- the state hash.
-hashBlockState :: (SupportsPersistentAccount pv m) => PersistentBlockState pv -> m (HashedPersistentBlockState pv)
+hashBlockState :: (SupportsPersistentAccount pv m, Modules.SupportsPersistentModules m) => PersistentBlockState pv -> m (HashedPersistentBlockState pv)
```

Comment:
I recommend defining
```
type SupportsPersistentState pv m = (SupportsPersistentAccount pv m, Modules.SupportsPersistentModules m)
```
or perhaps
```
type SupportsPersistentState pv m =
    ( IsProtocolVersion pv,
      MonadBlobStore m,
      MonadCache (AccountCache (AccountVersionFor pv)) m,
      MonadCache ModuleCache m
    )
```

Reply:
Yes. I was a bit unsure if it was worth the hazzle of introducing a new abstraction for this pair of constraints. 

However thinking twice I actually lean towards your suggestion of adding it, so I will introduce it.